src/main/scala/centaur/test/Test.scala

```diff
@@ -48,6 +47,9 @@ object Test {
         override def run: Try[A] = Try(x)
       }
     }
+
+    /** Call the default non-stack-safe but correct version of this method. */
+    override def tailRecM[A, B](a: A)(f: (A) => Test[Either[A, B]]): Test[B] = defaultTailRecM(a)(f)
```

Comment:
It's almost as if someone coded things in a way that would cause stack overflows 😨